pytest_html/plugin.py

```diff
@@ -157,24 +158,24 @@ def create_asset(self, content, extra_index,
         def append_extra_html(self, extra, extra_index, test_index):
             href = None
             if extra.get('format') == extras.FORMAT_IMAGE:
-                if self.self_contained:
+                content = extra.get('content')
+                if re.match('^file|^http', content):
+                    html_div = html.img(src=content)
+                elif self.self_contained:
```

Comment:
@davehunt I need some guidance on how to add the pytest warning. Too many options. Do I get the terminal reporter via getplugin()? Do I use one of the hooks? etc.